metrics.go

```diff
@@ -0,0 +1,86 @@
+package es
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v7/estransport"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var clientDurationSummary *prometheus.SummaryVec
+
+func init() {
+	clientDurationSummary = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: "es_search_operations_durations_seconds",
+			Help: "es_search_operations_durations_seconds",
+		},
+		[]string{"client_name", "operation", "status", "resp_code"},
+	)
+
+	prometheus.MustRegister(clientDurationSummary)
+}
+
+type EsTransportWithMetrics struct {
+	EsTransport estransport.Interface
+}
+
+func (t EsTransportWithMetrics) Perform(r *http.Request) (*http.Response, error) {
+	start := time.Now()
+
+	resp, err := t.Perform(r)
+
+	status := "ok"
+	if err != nil {
+		status = "error"
+	}
+
+	operation := "search"
+	if !strings.Contains(r.URL.Path, "_search") {
```

Comment:
lets put `r.URL.Path` as operation label value. this way we don't need to update this adapter in case if another ES API method will be used.

Reply:
done, now operation values will include index name and operation e.g. bookdb_index/book/_search